python/ray/train/huggingface/transformers/_transformers_utils.py

```diff
@@ -87,11 +89,21 @@ def process_dataset_for_hf(
         hf_iterable, format_type="torch"
     ).with_format("torch")
 
-    try:
-        dataset_length = dataset._base_dataset.count()
-    except (ValueError, AttributeError):
-        # pipeline case
-        dataset_length = None
+    if isinstance(dataset, StreamSplitDataIterator):
+        if isinstance(dataset._base_dataset, MaterializedDataset):
```

Comment:
@antoine-galataud @gjoliver

Reply:
Cool. I feel like we should show a warning message when we can’t compute the length. Telling users they need to provide # of train steps instead.